src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -59,7 +59,7 @@ typedef struct {
   int64_t task_counter;
   /** A queue of tasks to be executed on this actor. The tasks will be sorted by
    *  the order of their actor counters. */
-  task_queue_entry *task_queue;
+  std::list<TaskQueueEntry *> *task_queue;
```

Comment:
You should probably use `std::vector` instead of `std::list` as it has better data locality